TRACING.md

```diff
@@ -0,0 +1,272 @@
+
+Tracing support in the agent is either disabled, or set to one of the
+following:
+
+| Trace type | Trace setting | Description | Use-case | Notes |
+|-|-|-|-|-|
+| "isolated" | `agent.trace=isolated` (or just `agent.trace`) | The traces only apply to the agent; after the container has been destroyed, the first span will start at agent startup and the last at agent shutdown | Observing agent lifespan. | |
+| "collated" | `agent.trace=collated` | In this mode, spans are associated with their `kata-runtime` initiated counterparts. | Understanding how the runtime calls the agent. | Requires runtime tracing to be enabled in `configuration.toml` (`enable_tracing=true`). |
+
+For example, to enable isolated tracing explicitly, add the following to the
+runtime's `configuration.toml` file:
+
+```
+kernel_params = "agent.trace=isolated ... "
+```
+
+> **Note:**
+>
+> Agent tracing is separate from the tracing for other Kata Containers
+> components. It is not necessary to enable runtime tracing if you want to
+> enable agent tracing (or *vice versa*). However, "collated" mode only works
+> as documented if runtime tracing is enabled.
+
+# Running the agent with tracing enabled
+
+1. Build a custom non-initrd image using
+   [osbuilder](https://github.com/kata-containers/osbuilder).
+
+   - The image **must** include:
+     - The `jaeger-client-socat-redirector.service` systemd service
+       (see the [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock) section).
+
+     - The distribution-specific package including the `socat(1)` utility
+       (normally the package is called `socat`).
+
+       The utility is required for the `jaeger-client-socat-redirector.service` service.
+
+   - The image can include:
+     - The `kata-journald-host-redirect.service` service.
+       (see the [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock) section).
+     - The `kata-redirect-agent-output-to-journal.conf` systemd snippet
+       (see the [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock) section).
+
+1. Install the custom image.
+
+1. Configure the runtime to use the custom image by modifying the `image=`
+   option in the runtime's `configuration.toml` file.
+
+1. Configure the runtime to set the correct kernel parameters for agent tracing. See:
+
+   - The [Tracing options](#tracing-options) section.
+
+   - If you also want full debug and guest journal redirection, see the
+     [Enable full agent debug](#enable-full-agent-debug) section.
+
+1. Install and run Jaeger on the host.
+
+   The simplest method to start using Jaeger is to follow the instructions to run the
+   ["all-in-one" Docker\* image](https://www.jaegertracing.io/docs/1.8/getting-started/#all-in-one).
+
+   > **Note:**
+   >
+   > Currently, it is necessary to run the `all-in-one` image with `--runtime runc`.
+   >
+   > See: https://github.com/kata-containers/runtime/issues/942.
```

Comment:
issue 942 appears to be closed as 'not a problem' - do you need to update this item?

Reply:
Dropping...